### ISSUE TITLE
Corrupt fix-sweave-exit-code.patch at R

### ISSUE THREAD

**Issue by KaiLemke** [NixOS/nixpkgs#25572]
## Issue description
Installing R fails, because fix-sweave-exit-code.patch cannot be applied:

applying patch /nix/store/wyfr6an0xj2iiwa8fkrryyzrrv5wnvnk-fix-sweave-exit-code.patch
patching file src/library/utils/R/Sweave.R
Hunk #1 FAILED at 1.
Hunk #2 FAILED at 452.
Hunk #3 FAILED at 577.
3 out of 3 hunks FAILED -- saving rejects to file src/library/utils/R/Sweave.R.rej
builder for ‘/nix/store/y6n3klamjnqnw2mx6rlia5f7829amqnb-R-3.3.3.drv’ failed with exit code 1

### Steps to reproduce

add pkgs.R to configuration.nix
nixos-rebuild test  --show-trace

## Technical details

* System: NixOS: 17.03.1098.a162c6c828 (Gorilla), stable, 64 Bit
* Sandboxing enabled: false

**Comment by JayVii** [comment]
I can confirm. Have the same issue:
```
unpacking source archive /nix/store/ahvpik8cyjhgirc6zb14hhhc4l554lvh-R-3.3.3.tar.gz
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0source root is R-3.3.3
  setting SOURCE_DATE_EPOCH to timestamp 1488788159 of file R-3.3.3/src/library/utils/inst/doc/Sweave.pdf
  patching sources
  applying patch /nix/store/qnbw3qijmgcyldy30wf66bxyq5cwasbn-no-usr-local-search-paths.patch
  patching file configure
  Hunk #1 succeeded at 3731 (offset -69 lines).
  applying patch /nix/store/wyfr6an0xj2iiwa8fkrryyzrrv5wnvnk-fix-sweave-exit-code.patch
  patching file src/library/utils/R/Sweave.R
  Hunk #1 FAILED at 1.
  Hunk #2 FAILED at 452.
  Hunk #3 FAILED at 577.
  3 out of 3 hunks FAILED -- saving rejects to file src/library/utils/R/Sweave.R.rej
  builder for ‘/nix/store/y6n3klamjnqnw2mx6rlia5f7829amqnb-R-3.3.3.drv’ failed with exit code 1
  )
```

**Comment by pschuprikov** [comment]
Comparing [master](https://github.com/NixOS/nixpkgs/commits/master/pkgs/applications/science/math/R/default.nix) and [17.03](https://github.com/NixOS/nixpkgs/commits/release-17.03/pkgs/applications/science/math/R/default.nix) it seems that commit d16c38a260ef41ae648f994918f491437fb7c75c is missing from the latter. [Presumably](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=862283), the patch was intended for version 3.4.0. @peti, could you, please, check?

**Comment by peti** [comment]
Fixed by 05b2da80fe47ce2c54fa75f60f697db566bf1700.
